elk/training/__init__.py

```diff
@@ -1,4 +1,12 @@
-from .ccs_reporter import CcsReporter, CcsReporterConfig
-from .eigen_reporter import EigenReporter, EigenReporterConfig
-from .reporter import OptimConfig, Reporter, ReporterConfig
-from .train import RunConfig
+
+
+all = [
+    "Reporter",
+    "ReporterConfig",
+    "CcsReporter",
+    "CcsReporterConfig",
+    "EigenReporter",
+    "EigenReporterConfig",
+    "RunConfig",
+    "OptimConfig",
+]
```

Comment:
> this makes ruff automatically remove unused imports

but that also means that you need to explicitly export imports

Reply:
Oh this is a bit annoying actually...

ChatGPT suggests that you can automate this like so:
```
# Automatically fill the __all__ field
__all__ = [name for name, value in globals().items() if not name.startswith('__') and callable(value)]
```
appending this to the end of the `__init__` files

Reply:
ruff itself (and the user's own linters) wouldn't be able to understand that. That you've reeexported everything inside.
So whoever uses the elk pacakge will still get warnings, and ruff itself will warn us too. (False positive but annoying for everyone)


I've added 
```
ignore-init-module-imports = true
```
to the pyproject.toml. So what ruff will do is to tell us that we've imported something in `__init__.py` but didn't add it to`__all__`. That should fix future mistakes when we forget to declare it.